suzieq/poller/worker/nodes/node.py

```diff
@@ -479,205 +383,215 @@ async def _parse_device_type_hostname(self, output, _) -> None:
                 devtype = None
 
             if not devtype:
                 self.logger.debug(
                     f"no devtype for {self.hostname} {self.current_exception}")
-                self._status = "init"
                 return
             else:
-                self.set_devtype(devtype, '')
-                self._status = "good"
-
-                self.set_hostname(hostname)
-
-            await self.init_boot_time()
+                self._set_devtype(devtype, '')
+                self._set_hostname(hostname)
+        else:
+            self.logger.error(
+                f'Non-SSH transport node {self.address}:{self.port} '
+                'has no devtype specified. Node will not be polled')
 
-    async def get_device_type_hostname(self):
+    async def _get_device_type_hostname(self):
         """Determine the type of device we are talking to if using ssh/local"""
         # There isn't that much of a difference in running two commands versus
         # running them one after the other as this involves an additional ssh
         # setup time. show version works on most networking boxes and
         # hostnamectl on Linux systems. That's all we support today.
-        await self.exec_cmd(self._parse_device_type_hostname,
-                            ["show version", "hostname",
-                             "cat /etc/os-release", "show hostname"], None,
-                            'text')
+        await self._exec_cmd(self._parse_device_type_hostname,
+                             ["show version",
+                              "cat /etc/os-release && hostname"],
+                             None, 'text', only_one=True)
 
-    def set_devtype(self, devtype: str, version_str: str) -> None:
+    def _set_devtype(self, devtype: str, version_str: str) -> None:
         """Change the class based on the device type"""
 
-        self.devtype = devtype
         if not devtype:
             return
+
         if devtype not in known_devtypes():
             self.logger.error(f'An unknown devtype {devtype} is being added.'
                               f' This will cause problems. '
                               f'Node {self.address}:{self.port}')
             raise ValueError
 
-        if self.devtype == "cumulus":
-            self.__class__ = CumulusNode
-        elif self.devtype == "eos":
-            self.__class__ = EosNode
-        elif self.devtype == "iosxr":
-            self.__class__ = IosXRNode
-        elif self.devtype == "iosxe":
-            self.__class__ = IosXENode
-        elif self.devtype == "ios":
-            self.__class__ = IOSNode
-        elif self.devtype.startswith("junos"):
-            self.__class__ = JunosNode
-        elif self.devtype == "nxos":
-            self.__class__ = NxosNode
-        elif self.devtype.startswith("sonic"):
-            self.__class__ = SonicNode
-        elif self.devtype == "panos":
-            self.__class__ = PanosNode
+        if self.devtype != devtype:
```

Comment:
What means this line? When it happen that self.devtype != devtype? Only when `self.devtype == ''` I believe. In that case, we should change the if statement

Reply:
This means don't bother doing everything all over again if the devtype is already the correct type.